airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/AsyncStreamConsumer.java

```diff
@@ -204,7 +210,14 @@ public void close() throws Exception {
     flushWorkers.close();
 
     bufferManager.close();
-    onClose.accept(hasFailed);
+
+    final Map<StreamDescriptor, StreamSyncSummary> streamSyncSummaries = streamNames.stream().collect(toMap(
+        streamDescriptor -> streamDescriptor,
+        streamDescriptor -> new StreamSyncSummary(
+            Optional.of(recordCounts.get(streamDescriptor)).map(AtomicLong::get)
```

Comment:
Won't this work ? Optional.of(recordCounts.get(streamDescriptor).get()) I see you already defaulted it to AtomicLong(0) in constructor. so not present wouldn't be the case ?

Reply:
lol I was waffling between "preinitialize in the constructor" and "defensive handling when consuming". Yes, I should remove one of those - I'm inclined to keep the defensive handling here (+ switch it to a getOrDefault) and remove the preinit in the constructor, wdyt?

Reply:
pushed https://github.com/airbytehq/airbyte/pull/33232/commits/e24c5123069acba45a94cdfa3712aef8d0565ad8.

(also... my previous code wasn't even safe. It should have been Optional.ofNullable, at least >.>)

Reply:
ok this should _actually_ work now https://github.com/airbytehq/airbyte/pull/33232/commits/740ee1af634b1f9032c31ff741abf0eecbc1b927

(I always mix up the map get+default methods >.>)

Reply:
👍